esp-hal/src/clock/mod.rs

```diff
@@ -666,3 +670,64 @@ impl<'d> RadioClockController<'d> {
         clocks_ll::reset_rpa();
     }
 }
+
+#[derive(Clone, Copy)]
+#[doc(hidden)]
+/// These are all modems with bitmasks as their representation.
+pub enum ModemBitmask {
+    #[cfg(wifi)]
+    WiFi = 1,
+    #[cfg(bt)]
+    Bt = 2,
+    #[cfg(ieee802154)]
+    Ieee802154 = 4
+}
+
+/// Tracks which modems currently request an active PHY clock.
+static PHY_CLOCK_REFS: AtomicU8 = AtomicU8::new(0);
+
+
+// These functions are moved out of the trait to prevent monomorphization for every modem clock
+// controller. If that doens't really make sense, this can be moved back.
+
+fn enable_phy_clock_internal(modem_bitmask: ModemBitmask) {
+    if PHY_CLOCK_REFS.fetch_or(modem_bitmask as u8, Ordering::Relaxed) == 0 {
+        clocks_ll::enable_phy(true);
+    }
+}
+fn disable_phy_clock_internal(modem_bitmask: ModemBitmask) {
+    if PHY_CLOCK_REFS.fetch_and(!(modem_bitmask as u8), Ordering::Relaxed) == modem_bitmask as u8 {
+        clocks_ll::enable_phy(false);
```

Comment:
There is a race condition here, if code enters `enable_phy` and another clock user enables its clock, the result is unpredictable.

Reply:
Should I use a critical section in this case?